### ISSUE TITLE
unsupported hash type sha

### ISSUE THREAD

**Issue by gygabyte017** [python-poetry/poetry#7778]
- **Poetry version**: 1.4.2
- **Python version**: 3.8.16
- **OS version and name**: Windows Server 2019
- **pyproject.toml**: none

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue

When installing the package `swat` I get this error: `unsupported hash type sha`

```
>poetry add swat@latest
Using version ^1.12.1 for swat

Updating dependencies
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/3c/c8/5d6bb20ba20d8770e933d8c0fe956fdc11940af2c39f5b23e02196e0395c/swat-1.12.1-0-cp27-cp27m-macosx_10_9_x86_64.whl#sha256=30ad0ce7e69fa40e19433aa8327b55ae1c0f092fResolving dependencies... (1.6s)

Writing lock file

Package operations: 1 install, 0 updates, 0 removals

  • Installing swat (1.12.1): Failed

  ValueError

  unsupported hash type

  at D:\miniconda\lib\hashlib.py:157 in __hash_new
      153│         # It does neither support keyed blake2 nor advanced features like
      154│         # salt, personal, tree hashing or SSE.
      155│         return __get_builtin_constructor(name)(data, **kwargs)
      156│     try:
    → 157│         return _hashlib.new(name, data)
      158│     except ValueError:
      159│         # If the _hashlib module (OpenSSL) doesn't support the named
      160│         # hash, try using our builtin implementations.
      161│         # This allows for SHA224/256 and SHA384/512 support even though

The following error occurred when trying to handle this error:


  ValueError

  unsupported hash type sha

  at D:\miniconda\lib\hashlib.py:120 in __get_builtin_constructor
      116│     constructor = cache.get(name)
      117│     if constructor is not None:
      118│         return constructor
      119│
    → 120│     raise ValueError('unsupported hash type ' + name)
      121│
      122│
      123│ def __get_openssl_constructor(name):
      124│     if name in __block_openssl_constructor:
```

**Comment by dimbleby** [comment]
You've checked the box to declare that you included output with `-vvv`, but you haven't done so.

The `RECORDS` file in the swat wheel contains nonsense hashes.  Please report this to that project - https://github.com/sassoftware/python-swat/.

If you're feeling keen, please also report to https://github.com/pypa/installer/ that it would be nice if they handled this more gracefully

**Comment by abn** [comment]
Closing this issue as the external issue has been resolved and swat@1.13.1 is installable. No real action on Poetry side. Maybe improvement on error handling here - happy to review PRs submitted to this effect.

Relevant code that installs the wheel is at https://github.com/python-poetry/poetry/blob/9b3893a46f20d422ae1912109428b00bcce320b5/src/poetry/installation/wheel_installer.py#L88-L116.

The exception can be captured and handled here as well. https://github.com/python-poetry/poetry/blob/9b3893a46f20d422ae1912109428b00bcce320b5/src/poetry/installation/executor.py#L574
